### PR TITLE
raftstore: more condition checks before proposing raft cmd

### PR DESCRIPTION
**Pull request by 5kbpers**
Signed-off-by: 5kbpers <tangminghua@pingcap.com>

###  What have you changed?

This PR adds more condition checks before proposing a raft cmd.
fix https://github.com/tikv/tikv/issues/5875

###  What is the type of the changes?

- Bugfix (a change which fixes an issue)

###  How is the PR tested?

- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No.

###  Does this PR affect `tidb-ansible`?

No.

### CODE REVIEW

File: src/raftstore/store/fsm/peer.rs [hunk 1]
```diff
@@ -2341,6 +2341,18 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
             self.ctx.raft_metrics.invalid_proposal.mismatch_peer_id += 1;
             return Err(e);
         }
+        if !self.fsm.peer.is_initialized() {
```
Comment (Connor1996) [inline src/raftstore/store/fsm/peer.rs]: should check it in `read.rs` too?

Comment (Connor1996) [reply src/raftstore/store/fsm/peer.rs]: Also, please add comments

Comment (5kbpers) [reply src/raftstore/store/fsm/peer.rs]: This check should apply to all proposals, not only read.

Comment (Connor1996) [reply src/raftstore/store/fsm/peer.rs]: If the proposal is processed by the local_reader, it wouldn't call here.

Comment (5kbpers) [reply src/raftstore/store/fsm/peer.rs]: It also applies to write cmd if we enable follower to redirect proposals in the future. And the key problem is `LocalReader` does not have any information about peer.😥️

Comment (BusyJay) [reply src/raftstore/store/fsm/peer.rs]: How local reader handles follower read?

Comment (5kbpers) [reply src/raftstore/store/fsm/peer.rs]: follower read will potentially be rejected by two conditions in LocalReader:
1. `ReadDelegate.applied_index_term != ReadDelegate.term`. `applied_index_term` is  only updated after the leader applies a proposal, `term` is only updated when peer transfers to leader
2. lease is expired

Comment (BusyJay) [reply src/raftstore/store/fsm/peer.rs]: How can follower have lease?

Comment (5kbpers) [reply src/raftstore/store/fsm/peer.rs]: > How can follower have lease?

Then follower read will always be rejected and redirected to raftstore,

**Pull request merged**